### ISSUE TITLE
Add support for Julia

### ISSUE THREAD

**Issue by waynelapierre** [spyder-ide/spyder#7274]
I have used Spyder with Python for several years, and I really like its Matlab style features, I was wondering if you have considered adding support Julia.

**Comment by ccordoba12** [comment]
We have thought about that in the past, but:

1. It'd be a very big endeavor to support Julia right now.
2. We have very limited resources.
3. None of the core developers work with Julia.
4. There's [Juno](junolab.org) for Julia, which is very similar to Spyder but supported by Julia developers.

**Comment by CAM-Gerlach** [comment]
Of course, if you were willing/able to fund it, or could work on (and maintain) it yourself, we would be glad to consider it. If not, your best bet is probably finding someone/some people in the Julia community to work on or fund it, if there is either significant interest in the features/functionality Spyder provides over what Juno currently offers, and/or in an IDE that fully supported Python as well as Julia, or the desire to help grow Julia's still-small community by exposing hundreds of thousands of users to the language via an established IDE for the most popular current data science (and beyond) platform. 

Particularly on that second point, it seems Julia has a bit of a "chicken or the egg" problem right now, where the benefits over existing languages (Python, R, C++, Fortran) for many applications are considerable on their own, but just not quite enough (or perceived as such) to motivate large scale adoption vs. more familiar languages. users don't want to invest without the packages and ecosystem to support them, while package developers aren't as inclined to spend similar effort as Python on a language with a far smaller current community. Perhaps Spyder could help boostrap things? At least, you could pitch that as an idea...

**Comment by sylvaticus** [comment]
The comment of @CAM-Gerlach seems a bit inflammatory.

I landed on this page for the same reason as the op, wondering if Spyder could be used, either directly or wit minor hacking, as an IDE for Julia.

**Why Spyder ?** Because (a) it works with iPython, the specific Python Jupyter kernel, so, without knowing its internal, I suspected that may be not too difficult to use other Jupyter kernels as backends (Julia but not only.. R, bash and in general any Jupyter kernels); (b) Sometimes I work in Julia, sometimes in Python.. It would just be nice to  keep the same IDE

**Why not Juno ?** While feature-rich (probably even more than Spyder), Juno is based on Atom, that remains a (highly configurable) text editor. Lot of hacking was needed to make Atom being an IDE for Julia and it is still slow and buggy (my opinion). And.. if I try to use it as an IDE for Julia and Python at the same time, of course the different extensions don't speak well with each other, so for example trying to put a breakpoint on a julia script may result in the python extension trying to do that instead (and I guess the opposite, maybe depends on the order the extensions are loaded, but that's not the point of this post..)

**Comment by goanpeca** [comment]
Actually as part of Spyder 5 we will work toward making julia and more languages have support in spyder via plugins.

**Comment by ma-galan** [comment]
> This is not going to come anytime soon because we have our hands full just supporting Python. Sorry to disappoint you guys.

Is there any update on this? Any chance Julia will be supported in a near future?

**Comment by ccordoba12** [comment]
@Episkiliski, there are no updates and I don't think we'll work on this in the future due to the reasons I mentioned in https://github.com/spyder-ide/spyder/issues/7274#issuecomment-396306525.

**Comment by sylvaticus** [comment]
I think this thread can be closed now...
I commented back in 2020, but of course things changed, there is also VSCode Editor now and less of a need for the Julia support in the Spider editor (that remains a super nice editor)..

**Comment by ccordoba12** [comment]
> I think this thread can be closed now...

Yep, I think that's fare @sylvaticus. We don't have the resources nor the expertise to add support for Julia in Spyder.
